src/replay_stage.rs

```diff
@@ -84,103 +106,62 @@ impl ReplayStage {
-                    // issue that the bank state was still changed, leading to inconsistencies with the
-                    // leader as the leader currently should not be publishing erroneous transactions
-                    inc_new_counter_info!(
-                        "replicate-stage_failed_process_entries",
-                        (i - start_entry_index)
-                    );
-
-                    break;
-                }
-
-                if 0 == num_ticks_to_next_vote {
-                    if let Some(sender) = vote_blob_sender {
-                        send_validator_vote(bank, vote_account_keypair, &cluster_info, sender)
-                            .unwrap();
-                    }
-                }
-                let (scheduled_leader, _) = bank
-                    .get_current_leader()
-                    .expect("Scheduled leader should be calculated by this point");
-
-                // TODO: Remove this soon once we boot the leader from ClusterInfo
-                if scheduled_leader != current_leader {
-                    cluster_info.write().unwrap().set_leader(scheduled_leader);
-                }
-
-                if my_id == scheduled_leader {
-                    num_entries_to_write = i + 1;
-                    break;
-                }
-                start_entry_index = i + 1;
-                num_ticks_to_next_vote = BLOCK_TICK_COUNT;
+            let tick = &entries[0];
+            assert_eq!(
+                tick.transactions.len(),
+                0,
+                "expecting the first entry in a block to be a tick"
+            );
+            bank.init_fork(current_slot, &tick.id, base_slot)
+                .expect("init_checkpoint");
+            let res = bank.process_fork_entries(current_slot, &entries);
+            if res.is_err() {
+                inc_new_counter_info!("replicate-stage_failed_process_entries", 1);
             }
+            inc_new_counter_info!(
+                "replicate-transactions",
+                entries.iter().map(|x| x.transactions.len()).sum()
+            );
         }
 
-        // If leader rotation happened, only write the entries up to leader rotation.
-        entries.truncate(num_entries_to_write);
-        *last_entry_id = entries
-            .last()
-            .expect("Entries cannot be empty at this point")
-            .id;
-
-        inc_new_counter_info!(
-            "replicate-transactions",
-            entries.iter().map(|x| x.transactions.len()).sum()
-        );
-
-        let entries_len = entries.len() as u64;
-        // TODO: In line with previous behavior, this will write all the entries even if
-        // an error occurred processing one of the entries (causing the rest of the entries to
-        // not be processed).
-        if entries_len != 0 {
-            ledger_entry_sender.send(entries)?;
-        }
+        let now = Instant::now();
+        // TODO: make this work
```

Comment:
@rob-solana @carllin does this TOD make sense?

Reply:
yep